ui/tests/src/test/java/org/jboss/windup/addon/ui/WindupUpdateDistributionCommandTest.java

```diff
@@ -0,0 +1,291 @@
+                while(!addon.getStatus().isStarted()) {
+                    try
+                    {
+                        Thread.sleep(100);
+                    }
+                    catch (InterruptedException e)
+                    {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+    }
+    private void changeUiAddonDirectoryToBeOlder(File homeAddonsDir, String uiPreviousVersion)
+    {
+        File uiAddonDirectory = new File(homeAddonsDir.getAbsolutePath() + "/org-jboss-windup-ui-windup-ui-"
+                    + uiPreviousVersion.replaceAll("\\.", "-"));
+        File olderVersionAddon = new File(homeAddonsDir.getAbsolutePath() + "/org-jboss-windup-ui-windup-ui-" + WINDUP_OLD_VERSION.replaceAll("\\.", "-"));
+        if (!olderVersionAddon.exists())
+        {
+            olderVersionAddon.mkdir();
+        }
+        try
+        {
+            FileUtils.copyDirectory(uiAddonDirectory, olderVersionAddon);
+            makeUiAddonVersionOlder(homeAddonsDir.getPath());
+            deleteWholeDirectory(uiAddonDirectory);
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    private String getPreviousVersion(String homeAddonsDirPath)
+    {
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder;
+        try
+        {
+            dBuilder = dbFactory.newDocumentBuilder();
+            String oldUiAddonVersion = "";
+            File installedXml = new File(homeAddonsDirPath + "/installed.xml");
+            Document doc = dBuilder.parse(installedXml);
+            // this does not work properly
+            Element documentElement = doc.getDocumentElement();
+            NodeList childNodes = documentElement.getElementsByTagName("addon");
+            for (int i = 0; i <= childNodes.getLength() - 1; i++)
+            {
+                Element item = (Element) childNodes.item(i);
+                if (item.getNodeName().equals("addon"))
+                {
+                    String addonName = item.getAttribute("name");
+                    if (addonName.equals("org.jboss.windup.ui:windup-ui"))
+                    {
+                        oldUiAddonVersion = item.getAttribute("version");
+                        return oldUiAddonVersion;
+                    }
+                }
+            }
+        }
+        catch (ParserConfigurationException e)
+        {
+            e.printStackTrace();
+        }
+        catch (Exception e)
+        {
+            System.out.println("Why?");
+        }
+        return null;
+    }
+
+    private void makeUiAddonVersionOlder(String homeAddonsDirPath)
+    {
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder;
```

Comment:
You could use the forge xml-parser for this:

http://forge.jboss.org/addon/org.jboss.forge.addon:parser-xml

But this looks fine.